app/_layout.tsx

```diff
@@ -125,213 +150,30 @@ function AppContent() {
-            },
-            headerTintColor: isDark
-              ? darkColors.textPrimary
-              : lightColors.textPrimary,
-          }}
-        />
-        <Stack.Screen
-          name="(info)/cycle-phase-details"
-          options={{
-            headerShown: true,
-            headerTitle: t('info:screenTitles.todaysInsights'),
-            headerShadowVisible: true,
-            headerStyle: {
-              backgroundColor: isDark
-                ? darkColors.surface
-                : lightColors.surfaceVariant,
-            },
-            headerTintColor: isDark
-              ? darkColors.textPrimary
-              : lightColors.textPrimary,
-          }}
-        />
-        <Stack.Screen
-          name="(info)/cycle-length-info"
-          options={{
-            headerShown: true,
-            headerTitle: t('info:screenTitles.cycleLength'),
-            headerShadowVisible: false,
-            headerStyle: {
-              backgroundColor: isDark
-                ? darkColors.surface
-                : lightColors.surface,
-            },
-            headerTintColor: isDark
-              ? darkColors.textPrimary
-              : lightColors.textPrimary,
-          }}
-        />
-        <Stack.Screen
-          name="(info)/period-length-info"
-          options={{
-            headerShown: true,
-            headerTitle: t('info:screenTitles.periodLength'),
-            headerShadowVisible: false,
-            headerStyle: {
-              backgroundColor: isDark
-                ? darkColors.surface
-                : lightColors.surface,
-            },
-            headerTintColor: isDark
-              ? darkColors.textPrimary
-              : lightColors.textPrimary,
-          }}
-        />
-        <Stack.Screen
-          name="(info)/late-period-info"
-          options={{
-            headerShown: true,
-            headerTitle: t('info:screenTitles.latePeriod'),
-            headerShadowVisible: false,
-            headerStyle: {
-              backgroundColor: isDark
-                ? darkColors.surface
-                : lightColors.surface,
-            },
-            headerTintColor: isDark
-              ? darkColors.textPrimary
-              : lightColors.textPrimary,
-          }}
-        />
-        <Stack.Screen
-          name="(info)/prediction-info"
-          options={{
-            headerShown: true,
-            headerTitle: t('info:screenTitles.howPredictionsWork'),
-            headerShadowVisible: false,
-            headerStyle: {
-              backgroundColor: isDark ? darkColors.panel : lightColors.panel,
-            },
-            headerTintColor: isDark
-              ? darkColors.textPrimary
-              : lightColors.textPrimary,
-          }}
-        />
+        {dynamicScreens.map(screen => {
+          const titleValue = screen.titleKey?.includes(':') ? t(screen.titleKey) : screen.titleKey;
```

Comment:
If this is doing what I think it is, and it's determining whether to translate UI copy based on the presence of a `:`, then that's extremely brittle.